Response/JWTAuthenticationSuccessResponse.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Lexik\Bundle\JWTAuthenticationBundle\Response;
+
+use Symfony\Component\HttpFoundation\JsonResponse;
+
+/**
+ * Response sent on successful JWT authentication.
+ *
+ * @internal
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+final class JWTAuthenticationSuccessResponse extends JsonResponse
+{
+    /**
+     * The Json Web Token.
+     *
+     * Immutable property.
+     *
+     * @var string
+     */
+    private $token;
+
+    /**
+     * The response extra data
+     *
+     * @var array
+     */
+    private $extraData;
+
+    /**
+     * @param string $token     Json Web Token
+     * @param array  $extraData Extra data passed to the response body.
+     */
+    public function __construct($token, array $extraData = [])
+    {
+        $this->token     = $token;
+        $this->extraData = $extraData;
+
+        parent::__construct();
+
+        $this->setData();
+    }
+
+    /**
+     * Gets the Json Web Token.
+     *
+     * @return string
+     */
+    public function getToken()
+    {
+        return $this->token;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setExtraData(array $extraData = [])
+    {
+        $this->extraData = $extraData;
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getExtraData()
+    {
+        return $this->extraData;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @param array $data Unused parameter for risks limitation
+     *
+     * @internal
+     */
+    public function setData($data = [])
```

Comment:
Wow... I was gone to far, written too late in the night... This will be widely sufficient.